tests/e2e/system_test/test_cr_resources_validation.py

```diff
@@ -22,47 +22,93 @@ class TestCRRsourcesValidation(E2ETest):
+        ), f"Failed to create resource {cr_object_kind} from yaml file {cr_resource_yaml}, got result {res}"
 
-        network_fence_name = res[0].split()[0]
-        self.object_name_to_delete = network_fence_name
+        cr_resource_name = res[0].split()[0]
+        self.object_name_to_delete = cr_resource_name
 
-        network_fence_original_yaml = run_oc_command(
-            f"get {network_fence_name} -o yaml"
-        )
+        cr_resource_original_yaml = run_oc_command(f"get {cr_resource_name} -o yaml")
 
-        patches = {  # dictionary: patch_name --> patch
-            "apiVersion": {"apiVersion": "csiaddons.openshift.io/v1alpha2"},
-            "kind": {"kind": "newNetworkFence"},
-            "generation": '{"metadata": {"generation": 6456456 }}',
-            "creationTime": '{"metadata": {"creationTimestamp": "2022-04-24T19:39:54Z" }}',
-            "name": '{"metadata": {"name": "newName" }}',
-            "uid": '{"metadata": {"uid": "897b3c9c-c1ce-40e3-95e6-7f3dadeb3e83" }}',
-            "secret": '{"spec": {"secret": {"name": "new_secret"}}}',
-            "cidrs": '{"spec": {"cidrs": {"10.90.89.66/32", "11.67.12.42/32"}}}',
-            "fenceState": '{"spec": {"fenceState": "NotFenced"}}',
-            "driver": '{"spec": {"driver": "example.new_driver"}}',
-            "new_property": '{"spec": {"new_property": "new_value"}}',
-        }
+        # test that all non editable properties are really not editable
+        non_editable_properties_errors = {}
+        cr_resource_prev_yaml = cr_resource_original_yaml
+        for patch in patches_non_editable:
+            params = "'" + f"{patches_non_editable[patch]}" + "'"
+            command = f"oc -n openshift-storage patch {cr_resource_name} -p {params} --type merge"
+
+            temp_file = NamedTemporaryFile(
+                mode="w+", prefix="cr_resource_modification", suffix=".sh"
+            )
+            with open(temp_file.name, "w") as t_file:
+                t_file.writelines(command)
+            self.temp_files_list.append(temp_file.name)
+            run_cmd(f"chmod 777 {temp_file.name}")
+            logger.info(f"Trying to edit property {patch}")
+
+            try:
+                run_cmd(f"sh {temp_file.name}")
+                cr_resource_modified_yaml = run_oc_command(
+                    f"get {cr_resource_name} -o yaml"
+                )
+
+                if cr_resource_prev_yaml != cr_resource_modified_yaml:
+                    non_editable_properties_errors[patch] = cr_resource_modified_yaml
+                    # reset prev yaml to the modified one to track further modifications
+                    cr_resource_prev_yaml = cr_resource_modified_yaml
+            except (
+                CommandFailed
+            ):  # some properties are not editable and CommandFailed exception is thrown
+                continue  # just continue to the next property
+
+        if non_editable_properties_errors:
```

Comment:
would using `assert` be a better option here?

Reply:
Unfortunately, it seems not. If there are non_editable_properties_errors, we do a series of actions: logging different error messages etc. and then throwing the exception. So 'if' seems to be a more appropriate option here than 'assert' which just throws exception with the message

Reply:
Eventually fixed it. Thanks.